crates/host/src/wasmbus/mod.rs

```diff
@@ -2520,17 +2559,16 @@ impl Host {
             "handling put wrpc link definition"
         );
 
-        // Before we store the link, we need to ensure the configuration is resolvable
-        let provider_link = self
-            // TODO(#2407): This is going to fail every time if a secret is defined in a link.
-            // Since links can be created before the component/provider we need some ability to
-            // just check config
-            .resolve_link_config(interface_link_definition.clone(), None, None)
-            .await
-            .context("failed to resolve link config for provider")?;
+        self.validate_config(
```

Comment:
Super nit: Validate config should probably just take `IntoIterator` where the item is `AsRef<str>`. Would avoid the need for a clone and allocating a new Vec

Reply:
Going through this was able to use IntoIterator, but not avoid a clone of the name.